### ISSUE TITLE
Connection Issue after ubuntu 19.04 update

### ISSUE THREAD

**Issue by aeryl1999** [ShinjiruApp/Shinjiru#43]
Everything was working perfectly on Ubuntu 18.04 but after updating to Ubuntu(Disco 19.04) Shinjiru stopped connecting to anilist entirely. Tried completely removing qt and Shinjiru before rebuilding but it still won't authenticate properly with anilist.I noticed that the app is using a socket port 0 in. Tried manually changing the authentication link to use port 42069, shinjiru warned about invalid request and returned a `TLS initialization failed` error

I believe this could be an issue with the latest ubuntu version dropping support for `libssl1.0-dev` and forcing `libssl-dev` to be used which is not compatible with openssl1.0.2

Any workarounds for this one?

```
[2019-04-25 00:21:29.020] Debug: Loaded style: "/home/admin/.config/Kazakuri/Shinjiru/themes/Default.qss"
[2019-04-25 00:21:29.773] Warning: QSslSocket: cannot resolve CRYPTO_num_locks
[2019-04-25 00:21:29.774] Warning: QSslSocket: cannot resolve CRYPTO_set_id_callback
[2019-04-25 00:21:29.774] Warning: QSslSocket: cannot resolve CRYPTO_set_locking_callback
[2019-04-25 00:21:29.775] Warning: QSslSocket: cannot resolve ERR_free_strings
[2019-04-25 00:21:29.775] Warning: QSslSocket: cannot resolve EVP_CIPHER_CTX_cleanup
[2019-04-25 00:21:29.775] Warning: QSslSocket: cannot resolve EVP_CIPHER_CTX_init
[2019-04-25 00:21:29.776] Warning: QSslSocket: cannot resolve sk_new_null
[2019-04-25 00:21:29.776] Warning: QSslSocket: cannot resolve sk_push
[2019-04-25 00:21:29.776] Warning: QSslSocket: cannot resolve sk_free
[2019-04-25 00:21:29.777] Warning: QSslSocket: cannot resolve sk_num
[2019-04-25 00:21:29.777] Warning: QSslSocket: cannot resolve sk_pop_free
[2019-04-25 00:21:29.777] Warning: QSslSocket: cannot resolve sk_value
[2019-04-25 00:21:29.777] Warning: QSslSocket: cannot resolve SSL_library_init
[2019-04-25 00:21:29.778] Warning: QSslSocket: cannot resolve SSL_load_error_strings
[2019-04-25 00:21:29.778] Warning: QSslSocket: cannot resolve SSL_get_ex_new_index
[2019-04-25 00:21:29.778] Warning: QSslSocket: cannot resolve SSLv3_client_method
[2019-04-25 00:21:29.779] Warning: QSslSocket: cannot resolve SSLv23_client_method
[2019-04-25 00:21:29.779] Warning: QSslSocket: cannot resolve SSLv3_server_method
[2019-04-25 00:21:29.779] Warning: QSslSocket: cannot resolve SSLv23_server_method
[2019-04-25 00:21:29.780] Warning: QSslSocket: cannot resolve X509_STORE_CTX_get_chain
[2019-04-25 00:21:29.780] Warning: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_noconf
[2019-04-25 00:21:29.780] Warning: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_conf
[2019-04-25 00:21:29.781] Warning: QSslSocket: cannot resolve SSLeay
[2019-04-25 00:21:29.781] Warning: Incompatible version of OpenSSL
[2019-04-25 00:21:29.802] Debug: File not found "/home/admin/.local/bin/customMediaPlayers.json"
[2019-04-25 00:21:30.174] Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed
[2019-04-25 00:21:30.610] Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed
[2019-04-25 00:21:30.611] Debug: "TLS initialization failed"
[2019-04-25 00:21:50.136] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.137] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.139] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.140] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.142] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.144] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.145] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.148] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.149] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
[2019-04-25 00:21:50.150] Warning: Invalid request: http://127.0.0.1:42069/favicon.ico
```

**Comment by Kazakuri** [comment]
Which version of Qt do you have installed?

As far as I know, support for OpenSSL v1.1 was added in Qt 5.10 but there was [some issues with it in Linux until Qt 5.12.1](https://wiki.qt.io/Qt_5.12.0_Known_Issues#Linux)

**Comment by aeryl1999** [comment]
I have the latest release of Qt(5.12.3) installed by default. Will try downgrading to 5.10 to see if it resolves this

**Comment by aeryl1999** [comment]
@Kazakuri Downgraded to Qt(5.11.1) Shinjiru won't start anymore

**Comment by aeryl1999** [comment]
Qt 5.12.0 fixed the launching issue but it's still showing an opennssl issue
```
[2019-04-25 08:55:13.576] Debug: Loaded style: "/home/admin/.config/Kazakuri/Shinjiru/themes/Default.qss"
[2019-04-25 08:55:13.895] Warning: QSslSocket: cannot resolve CRYPTO_num_locks
[2019-04-25 08:55:13.897] Warning: QSslSocket: cannot resolve CRYPTO_set_id_callback
[2019-04-25 08:55:13.897] Warning: QSslSocket: cannot resolve CRYPTO_set_locking_callback
[2019-04-25 08:55:13.897] Warning: QSslSocket: cannot resolve ERR_free_strings
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve EVP_CIPHER_CTX_cleanup
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve EVP_CIPHER_CTX_init
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve sk_new_null
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve sk_push
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve sk_free
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve sk_num
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve sk_pop_free
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve sk_value
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve SSL_library_init
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve SSL_load_error_strings
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve SSL_get_ex_new_index
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve SSLv3_client_method
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve SSLv23_client_method
[2019-04-25 08:55:13.898] Warning: QSslSocket: cannot resolve SSLv3_server_method
[2019-04-25 08:55:13.899] Warning: QSslSocket: cannot resolve SSLv23_server_method
[2019-04-25 08:55:13.899] Warning: QSslSocket: cannot resolve X509_STORE_CTX_get_chain
[2019-04-25 08:55:13.899] Warning: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_noconf
[2019-04-25 08:55:13.899] Warning: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_conf
[2019-04-25 08:55:13.899] Warning: QSslSocket: cannot resolve SSLeay
[2019-04-25 08:55:13.899] Warning: Incompatible version of OpenSSL
[2019-04-25 08:55:13.925] Debug: File not found "/home/admin/.local/bin/customMediaPlayers.json"
[2019-04-25 08:55:13.946] Debug: QNetworkReplyHttpImplPrivate::_q_startOperation was called more than once QUrl("https://shinjiru.me/static/changes.json")
[2019-04-25 08:55:13.970] Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed
[2019-04-25 08:55:14.791] Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed
[2019-04-25 08:55:14.791] Debug: "TLS initialization failed"
[2019-04-25 08:55:19.746] Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed
[2019-04-25 08:55:19.746] Warning: TLS initialization failed
[2019-04-25 08:55:20.013] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.015] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.016] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.019] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.022] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.025] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.028] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.031] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.036] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:20.038] Warning: Invalid request: http://localhost:42069/favicon.ico
[2019-04-25 08:55:41.489] Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed
```

**Comment by Kazakuri** [comment]
Reading through some of the QTBUGs, it appears like they plan to [keep using OpenSSL 1.0.2 in their builds until it's EOL at the end of the year](https://bugreports.qt.io/browse/QTBUG-74315).

The only real solutions available if Ubuntu stopped shipping 1.0.2 would be either building OpenSSL 1.0.2 for yourself, or building Qt linked against OpenSSL 1.1.1 for yourself. Neither of which sound remotely ideal.

With that said, I have been making progress towards using Snap for Linux releases in #38 so it might be possible to distribute the correct version of OpenSSL with that and prevent this whole dependency nonsense to begin with.

**Comment by aeryl1999** [comment]
I guess I'll just have to wait for the snap
